flink-examples/src/main/java/io/pravega/examples/flink/alert/HighCountAlerter.java

```diff
@@ -0,0 +1,206 @@
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.util.Collector;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/*
+ * This application has the following input parameters
+ *     stream - Pravega stream name to write to
+ *     controller - the Pravega controller URI, e.g., tcp://localhost:9090
+ *                  Note that this parameter is processed in pravega flink connector
+ */
+public class HighCountAlerter {
+
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(HighCountAlerter.class);
+
+    // The application reads data from specified Pravega stream and once every ALERT_INTERVAL (2 seconds)
+    // counts the number of 500 responses in the last ALERT_WINDOW (30 seconds), and generates
+    // alert when the counts exceed ALERT_THRESHOLD (6).
+
+    public static void main(String[] args) throws Exception {
+        LOG.info("Starting HighErrorAlerter...");
+
+        // initialize the parameter utility tool in order to retrieve input parameters
+        ParameterTool params = ParameterTool.fromArgs(args);
+
+        // create Pravega helper utility for Flink using the input paramaters
+        FlinkPravegaParams helper = new FlinkPravegaParams(params);
+
+        // get the Pravega stream from the input parameters
+        StreamId streamId = helper.getStreamFromParam(Constants.STREAM_PARAM,
+                                                      Constants.DEFAULT_STREAM);
+
+        // create the Pravega stream is not exists.
+        helper.createStream(streamId);
+
+        // initialize Flink execution environment
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        // create the Pravega stream reader
+        long startTime = 0;
+        FlinkPravegaReader<String> reader = helper.newReader(streamId, startTime, String.class);
+
+        // add the Pravega reader as the data source
+        DataStream<String> inputStream = env.addSource(reader);
+
+        // create an output sink to stdout for verification
+        //inputStream.print();
+
+        // transform logs
+        DataStream<AccessLog> dataStream = inputStream.map(new ParseLogData());
+
+        // create an output sink to stdout for verification
+        //dataStream.print();
+
+        // get responses and their counts
+        DataStream<ResponseCount> countStream =
+            dataStream.flatMap(new FlatMapFunction<AccessLog, ResponseCount>() {
+                @Override
+                public void flatMap(AccessLog value, Collector<ResponseCount> out) throws Exception {
+                    out.collect(new ResponseCount(value.getStatus(), 1));
+                }
+            }).filter((FilterFunction<ResponseCount>) count -> {
```

Comment:
do we really need this filter?

Reply:
It was meant to deal with malformed data. No longer relevant. replaced with filter for 500 responses.